tfhe/src/integer/gpu/server_key/radix/ilog2.rs

```diff
@@ -91,12 +93,14 @@ impl CudaServerKey {
             },
         );
 
-        let mut output_cts = ct.duplicate_async(streams);
+        let mut output_cts: T =
+            self.create_trivial_zero_radix_async(num_ct_blocks * num_ct_blocks, streams);
```

Comment:
This is the main fix, I had to modify  `compute_prefix_sum_hillis_steele_async` to take a range as well.